trl/trainer/utils.py

```diff
@@ -75,21 +75,31 @@ class DataCollatorForCompletionOnlyLM(DataCollatorForLanguageModeling):
     def __init__(
         self,
         response_template: Union[str, List[int]],
-        instruction_template: Optional[str] = None,
+        instruction_template: Union[str, List[int]] = None,
         *args,
         mlm: bool = False,
         ignore_index: int = -100,
         **kwargs,
```

Comment:
```suggestion
        *args,
        mlm: bool = False,
        ignore_index: int = -100,
        instruction_template: Union[str, List[int]] = None,
        **kwargs,
```
To preserve previous behaviour I propose to change the order of the arguments, what do you think?

Reply:
Won't this break existing code that uses this API like so?


```python
DataCollatorForCompletionOnlyLM(response_template, instruction_template)
```

Reply:
E.g. it would break this test 

https://github.com/devxpy/trl/blob/8324c05405f87815000f0dc23598ad5ce54fd939/tests/test_data_collator_completion_only.py#L61-L64

Reply:
Ah correct, yes, your proposed changes make sense